doc/developer-guides/trusty.rst

```diff
@@ -0,0 +1,186 @@
+.. _trusty_tee:
+
+Trusty TEE on ACRN
+##################
+
+Introduction
+************
+
+`Trusty`_ is a set of software components supporting a Trusted Execution Environment (TEE).
```

Comment:
There's a (bit) assumption here that the reader knows what a TEE is.  Feels like a summary of TEE would be good here,  and a pointer to where folks could read more about it, maybe here: https://datatracker.ietf.org/wg/teep/about/  or here: https://en.wikipedia.org/wiki/Trusted_execution_environment

Reply:
I'll add one